src/Analyzers/CSharp/Analyzers/UseExpressionBodyForLambda/UseExpressionBodyForLambdaDiagnosticAnalyzer.cs

```diff
@@ -31,33 +31,17 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
             => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(AnalyzeIfEnabled,
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax,
                 SyntaxKind.SimpleLambdaExpression, SyntaxKind.ParenthesizedLambdaExpression);
 
-        private static void AnalyzeIfEnabled(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
         {
             var analyzerOptions = context.Options;
             var syntaxTree = context.SemanticModel.SyntaxTree;
             var optionValue = UseExpressionBodyForLambdaHelpers.GetCodeStyleOption(analyzerOptions.GetAnalyzerOptions(syntaxTree));
-            var severity = UseExpressionBodyForLambdaHelpers.GetOptionSeverity(optionValue);
-            switch (severity)
-            {
-                case ReportDiagnostic.Error:
-                case ReportDiagnostic.Warn:
-                case ReportDiagnostic.Info:
-                    break;
-                default:
-                    // don't analyze if it's any other value.
-                    return;
-            }
```

Comment:
This is the main change. We should still analyze even if the option is of hidden diagnostic.